### DISCUSSION TITLE
@responsive directive doesn't work with postcss-each plugin

### DISCUSSION THREAD

**Discussion by n3ps** [tailwindlabs/tailwindcss#6093] [Help]
**What version of Tailwind CSS are you using?**
v2.2.19

**What build tool (or framework if it abstracts the build tool) are you using?**
Next.js 12.0.3

**What version of Node.js are you using?**
v14.18.1

**What browser are you using?**
Chrome

**What operating system are you using?**
macOS

**Reproduction steps**
1. npx create-next-app -e with-tailwindcss my-project
2. yarn add postcss-each and include it in postcss.config.js
3. create a styles.css with the following and import in _app.js:
```
@tailwind base;
@tailwind components;
@tailwind utilities;

@responsive {
  @each $item in red, green, blue {
    .$(item) {
      background: $(item);
    }
  }
}
```

**Describe your issue**
**Without** the `@responsive` directive, the `@each` postcss directive works. 

**With** the `@responsive` directive, it doesn't work. It silently ignores what's inside and the selectors are no longer generated.

**Comment by adamwathan** [comment]
Hey! You didn't provide your `postcss.config.js` file here but I'm guessing the issue is you are using `postcss-each` _after_ Tailwind, when it needs to come before. Tailwind doesn't understand the `@each` syntax so you need that code to run first so that by the time Tailwind sees it, it's just a static list of classes.

**Reply by n3ps** [reply]
`postcss-each` is registered before Tailwind and it works alongside "common" Tailwind usage. 

It only breaks when used _inside_ of `@responsive`. I also tried the `@variants` directive.
